Source/WebCore/style/RuleFeature.cpp

```diff
@@ -79,6 +79,34 @@ bool isHasPseudoClassMatchElement(MatchElement matchElement)
     }
 }
 
+static bool isScopeBreaking(MatchElement matchElement)
+{
+    switch (matchElement) {
+    case MatchElement::HasAnySibling:
+    case MatchElement::HasNonSubjectOrScopeBreaking:
+        return true;
```

Comment:
We should split HasNonSubjectOrScopeBreaking into two separate cases. HasNonSubject does not need scope breaking handling.